### PR TITLE
[docs] Revise installation guide

### LINKED ISSUE #1894: Improvements to the Solver Table
I had an idea to improve the current solver table.

It will be beneficial for first-time users if we add a column to the solver table which tells us which solvers work "out of the box" and which ones need a license or external installation. While the current table does have a column for license which is partly meant to indicate this, it could be made more explicit. This is also a common point in the solver specific notes and should probably be tabulated.

### PR DESCRIPTION
**Pull request by odow**
A few, mostly re-ordering changes to the installation guide. For example, showing how to install a solver up-front, rather than burying it in a paragraph after the table, and using `Clp` as an example instead of `Gurobi` so that everybody should be able to install it.

I added a "Manual Installation" column (#1894), simplified the Supports column, added a ᴹ for the MATLAB solvers, and removed the double-link for Julia solvers.. Here's what it looks like:
image
It scales nicely as the horizontal width of the page is resized.

Closes #2140 
Closes #2075
Closes #1894

### CODE REVIEW

File: docs/src/installation.md [hunk 1]
```diff
@@ -1,85 +1,128 @@
...
--   MILP = Mixed-integer linear programming
--   NLP = Nonlinear programming
--   MINLP = Mixed-integer nonlinear programming
--   SDP = Semidefinite programming
--   MISDP = Mixed-integer semidefinite programming
-
-You may also use [AmplNLWriter](https://github.com/jump-dev/AmplNLWriter.jl) to
-access solvers that support the [nl format](https://en.wikipedia.org/wiki/Nl_(format)).
+!!! note
+    Developed a solver? This table is open for new contributions! Start by
```
Comment (mlubin) [inline docs/src/installation.md:105]: Maybe "solver or solver wrapper"

File: docs/src/installation.md [hunk 2]
```diff
@@ -1,85 +1,128 @@
...
+Use [AmplNLWriter](https://github.com/jump-dev/AmplNLWriter.jl) to access
+solvers that support the [nl format](https://en.wikipedia.org/wiki/Nl_(format)).
 Such solvers include [Bonmin](https://github.com/coin-or/Bonmin) and
 [Couenne](https://github.com/coin-or/Couenne). See a more complete list
 [here](https://ampl.com/products/solvers/all-solvers-for-ampl/).
 
-Additionally, you may use [GAMS](https://www.gams.com) (Julia package:
-[GAMS.jl](https://github.com/GAMS-dev/gams.jl)) to access additional solvers. Among them are
+Use [GAMS.jl](https://github.com/GAMS-dev/gams.jl) to access solvers via an
+installation of [GAMS](https://www.gams.com). Among them are:
```
Comment (mlubin) [inline docs/src/installation.md:125]: Use GAMS.jl to access solvers available through GAMS. Among them are:

File: docs/src/installation.md [hunk 3]
```diff
@@ -1,85 +1,128 @@
...
-Getting Solvers
----------------
+JuMP depends on solvers to solve optimization problems, and you will need to
+install one before you can solve problems with JuMP. The table below lists the
+currently available solvers.
 
-JuMP depends on solvers to solve optimization problems. Most solvers are not
-written in Julia, and some require commercial licenses to use, so installation
-is often more complex. We list below the currently available solvers.
+Common choices of solvers are [Clp.jl](https://github.com/jump-dev/Clp.jl) for
```
Comment (mlubin) [inline docs/src/installation.md:20]: I'd put solver recommendations aside for a separate PR. JuMP hasn't officially done this before, so it needs some discussion.

File: docs/src/installation.md [hunk 4]
```diff
@@ -1,85 +1,124 @@
...
+| [Juniper.jl](https://github.com/lanl-ansi/Juniper.jl)                          |                                                                                  |     | MIT      | (MI)SOCP, (MI)NLP               |
+| [MOSEK](https://www.mosek.com/)                                                | [MosekTools.jl](https://github.com/jump-dev/MosekTools.jl)                       | Yes | Comm.    | (MI)LP, (MI)SOCP, SDP           |
+| [NLopt](https://github.com/stevengj/nlopt)                                     | [NLopt.jl](https://github.com/JuliaOpt/NLopt.jl)                                 |     | GPL      | LP, QP, NLP                     |
+| [OSQP](https://osqp.org/)                                                      | [OSQP.jl](https://github.com/oxfordcontrol/OSQP.jl)                              |     | Apache   | LP, QP                          |
+| [Pavito.jl](https://github.com/jump-dev/Pavito.jl)                             |                                                                                  |     | MPL-2    | (MI)CP                          |
```
Comment (joaquimg) [inline docs/src/installation.md:81]: CP is not defined. It should be (MI)NLP.
The additional info cam state that it works globally if nlp is convex.

**Comment by joaquimg** [comment]
What of the column was Installation and the value "Manual" instead of yes? Maybe slightly thinner the better? Or it would be more confusing?

**Comment by joaquimg** [comment]
Nice troubleshooting page!
Another common check for user would be:
Run ] status
And check if the versions displayed match the ones you are of Manual or README of packages.
(And remember to post this info when opening a issue in JuMP and discourse)

File: docs/src/installation.md [hunk 5]
```diff
@@ -1,216 +1,313 @@
...
+
+!!! tip
+    When in doubt, run `import Pkg; Pkg.update()` to see if updating your
+    packages fixes the issue. Remember you will need to exit Julia and start a
+    new session for the changes to take effect.
+
+
+### Check the version of your packages
+
+Each package is versionsed with a [three-part number](https://semver.org) like
```
Comment (rtwalker) [inline docs/src/installation.md:264]: s/versionsed/versioned

**Comment by odow** [comment]
Bump. Any more changes? Otherwise I will merge; it's an improvement over the present.

**Review by blegat — APPROVED** [review]
